oak_functions/util/src/sync.rs

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use lock_api::{GuardSend, RawMutex};
+use std::{
+    hint::spin_loop,
+    sync::atomic::{AtomicBool, Ordering},
+};
+
+pub struct SpinLock(AtomicBool);
+
+unsafe impl RawMutex for SpinLock {
+    type GuardMarker = GuardSend;
+
+    const INIT: SpinLock = SpinLock(AtomicBool::new(false));
+
+    fn lock(&self) {
+        while let Err(_) =
```

Comment:
I would prefer consistency with the format of `try_lock`: `while !self.0.compare_exchange(false, true, Ordering::Acquire, Ordering::Relaxed).is_ok()`

Reply:
Done.